src/api/fetch.js

```diff
@@ -0,0 +1,78 @@
+/* @flow */
+
+import FormData from 'form-data';
+import fetch from 'node-fetch';
+
+import HttpError from './HttpError';
+
+type File = {
+  fieldname: string,
+  originalname: string,
+  buffer: Buffer,
+};
+
+export type RequestOptions = {
+  method: string,
+  url: string,
+  data?: ?mixed,
+  files?: File[],
+  headers?: { [string]: string },
+};
+
+type Init = {
+  method: string,
+  headers: { [string]: string },
+  body?: string | FormData,
+};
+
+export default async function request<T>({
+  method,
+  url,
+  data,
+  headers,
+  files,
+}: RequestOptions): Promise<?T> {
+  const init: Init = {
+    method,
+    headers: {
+      Accept: 'application/json',
+      ...headers,
+    },
+  };
+
+  if (data) {
+    if (files) {
+      const formData = new FormData();
+
+      Object.entries(data).forEach(([name, value]) =>
+        formData.append(name, value),
+      );
+
+      files.forEach(({ fieldname, buffer, originalname }) => {
+        formData.append(fieldname, buffer, originalname);
+      });
+
+      init.body = formData;
+    } else {
+      init.headers['Content-Type'] = 'application/json';
+      init.body = JSON.stringify({ data });
+    }
+  }
+  const response = await fetch(url, init);
+
+  if (
```

Comment:
we probably want to put this in the wrapper as well, since it would keep us from seeing headers on 204s/404s